x-pack/plugins/ml/server/shared_services/shared_services.ts

```diff
@@ -174,9 +192,38 @@ function getRequestItemsProvider(
     );
 
     if (clusterClient === null) {
-      throw new MLClusterClientUninitialized(`ML's cluster client has not been initialized`);
+      throw getCustomError(
+        'MLClusterClientUninitialized',
+        `ML's cluster client has not been initialized`
+      );
     }
 
+    const uiSettingsClient = getUiSettings()?.asScopedToClient(savedObjectsClient);
+    if (uiSettingsClient === null) {
+      throw getCustomError(
+        'MLUISettingsClientUninitialized',
+        `ML's UI settings client has not been initialized`
+      );
+    }
+
+    const getFieldsFormatRegistry = async () => {
+      let fieldFormatRegistry;
+      try {
+        fieldFormatRegistry = await getFieldsFormat()!.fieldFormatServiceFactory(uiSettingsClient!);
```

Comment:
This doesn't address the comment I originally made that it's asserting that `getFieldsFormat()` and `getUiSettings` return a non-null values.
As with `getClusterClient` it would be safer to just check that it is not null before continuing.

Below is what i was thinking.
Note, the last two lines are the same as your original commit, it's safe to use  `uiSettings` and `fieldsFormat` as we know they're not 'null`

```typescript
    const clusterClient = getClusterClient();
    if (clusterClient === null) {
      throw getCustomError(
        'MLClusterClientUninitialized',
        `ML's cluster client has not been initialized`
      );
    }
    const uiSettings = getUiSettings();
    if (uiSettings === null) {
      throw getCustomError(
        'MLUISettingsClientUninitialized',
        `ML's UI settings client has not been initialized`
      );
    }
    const fieldsFormat = getFieldsFormat();
    if (fieldsFormat === null) {
      throw getCustomError(
        'MLFieldFormatRegistryUninitialized',
        `ML's field format registry has not been initialized`
      );
    }

    const jobSavedObjectService = jobSavedObjectServiceFactory(
      savedObjectsClient,
      internalSavedObjectsClient,
      spaceEnabled,
      authorization,
      isMlReady
    );

    const uiSettingsClient = uiSettings.asScopedToClient(savedObjectsClient);
    const getFieldsFormatRegistry = () => fieldsFormat.fieldFormatServiceFactory(uiSettingsClient);
```

Reply:
Updated the `uiSettingsClient` check for undefined in https://github.com/elastic/kibana/pull/109073/commits/124768fd36fcd8fb2f6575a40018c4634c12a394